tests/unit/Core/Framework/Rule/RuleComparisonTest.php

```diff
@@ -151,4 +151,202 @@ public function testNumericComparisonThrowsExceptionIfUnsupportedOperatorIsUsed(
 
         RuleComparison::numeric(1.0, 1.0, 'unsupported');
     }
+
+    #[DataProvider('valuesForDateTimeComparison')]
+    public function testDateTimeComparison(\DateTime $itemValue, \DateTime $ruleValue, bool $result, string $operator): void
+    {
+        static::assertEquals($result, RuleComparison::datetime($itemValue, $ruleValue, $operator));
```

Comment:
please use `assertSame` where possible instead, as it is type safe.
please check all occurrences